ivy/functional/backends/paddle/layers.py

```diff
@@ -88,7 +92,29 @@ def conv1d(
     dilations: Union[int, Tuple[int]] = 1,
     out: Optional[paddle.Tensor] = None,
 ) -> paddle.Tensor:
-    raise IvyNotImplementedException()
+    if data_format == "NCW":
+        x = paddle.transpose(x, perm=[0, 2, 1])
+
+    df = "NCL"
+    filters = paddle.transpose(filters, perm=[2, 1, 0])
+    x_pad = _pad_before_conv(
+        x, filters, strides, padding, dims=1, dilations=dilations, data_format=df
+    )
+    padding = "VALID"
```

Comment:
this is not correct, the function takes in a `padding` argument and you should accept it as is or manipulate it to achieve the same result

Reply:
Hi @MahmoudAshraf97 , thanks for the comments:

The transposition needs to be done to match the shape gound-truth as it's done similarly in torch, tf, and numpy. The padding="Valid" is passed because the x input is already padded in the function `__pad_before_conv`, so there is no need to send the raw inputs; we can utilize the "Valid" option.

As for the hypotheses test, it passes all previous frameworks done already except paddle, and it's not good to change it for only paddle.

My first thought-out implementation was the code below:
```python

   strides = [strides] if isinstance(strides, int) else strides
   dilations = [dilations] if isinstance(dilations, int) else dilations
   df = "NCL" if data_format == "NCW" else "NLC"
   
    if isinstance(padding, int):
        padding = [(padding, padding)]
        
    x = _pad_before_conv(
        x, filters, strides, padding, dims=1, dilations=dilations, data_format=df
    )

    return paddle.nn.functional.conv1d(
        x=x,
        weight=filters,
        data_format=df,
        stride=strides,
        padding=padding,
        dilation=dilations,
    )
```
The `_pad_before_conv()` manipulates the padding, which could be used instead of the "valid" but it does not change anything.

will it be okay to chat further on Discord?

Reply:
I see your point, sure feel free to continue the discussion there, my ID is xmrfake